src/java/test/org/apache/zookeeper/server/ServerCxnExceptionsTest.java

```
@@ -0,0 +1,170 @@
+
+  private void NettySetup() throws Exception {
+    System.setProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN_FACTORY,
+      "org.apache.zookeeper.server.NettyServerCnxnFactory");
+    System.setProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN, "org.apache.zookeeper.server.MockNettyServerCnxn");
+    System.setProperty("exception.type", "NoException");
+  }
+
+  private void NIOSetup() throws Exception {
+    System.setProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN_FACTORY,
+      "org.apache.zookeeper.server.NIOServerCnxnFactory");
+    System.setProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN, "org.apache.zookeeper.server.MockNIOServerCnxn");
+    System.setProperty("exception.type", "NoException");
+  }
+
+  @AfterClass
+  public static void tearDownAfterClass() throws Exception {
+    System.clearProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN_FACTORY);
+    System.clearProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN);
+    System.clearProperty("exception.type");
+  }
+
+  @Test (timeout = 60000)
+  public void testNettyIOException() throws Exception {
+    tearDown();
+    NettySetup();
+    testIOExceptionHelper();
+  }
+
+  @Test (timeout = 60000)
+  public void testNIOIOException() throws Exception {
+    tearDown();
+    NIOSetup();
+    testIOExceptionHelper();
+  }
+
+  private void testIOExceptionHelper() throws Exception {
+    System.setProperty("exception.type", "IOException");
+    super.setUp();
+    final ZooKeeper zk = createClient();
+    final String path = "/a";
+    try {
+      // make sure zkclient works
+      zk.create(path, "test".getBytes(), ZooDefs.Ids.OPEN_ACL_UNSAFE,
+        CreateMode.EPHEMERAL);
+      fail("Should not come here");
+      Stat stats = zk.exists(path, false);
+      if (stats != null) {
+        int length = stats.getDataLength();
+      }
+    } catch (KeeperException.ConnectionLossException cle) {
+      LOG.info("ConnectionLossException: {}", cle);
+    } finally {
+      zk.close();
```

Aren't we missing a ``try-catch`` around ``zk.close()`` with an explicit comment to ignore any error message?